### PR TITLE
oauth: handle redirection through ?next

### PR DESCRIPTION
**Pull request by remi-dupre**

### CODE REVIEW

File: prologin/users/auth_token_views.py [hunk 1]
```diff
@@ -81,18 +81,18 @@ def get_url(self, *args, **kwargs):
...
         AuthToken.garbage_collect()
+        redirect_data = {'code': auth_token.code, 'state': state}
 
-        return client.redirect_url + '?' + urlencode(
-            {
-                'code': auth_token.code,
-                'state': state
-            })
+        if 'next' in self.request.GET:
+            redirect_data.update({'next': self.request.GET.get('next')})
```
Comment (zopieux) [inline prologin/users/auth_token_views.py]: What's wrong with `dict['next'] = value`?

File: prologin/users/auth_token_views.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ class AuthorizeView(PermissionRequiredMixin, RedirectView):
     client_id_query = 'client_id'
     state_query = 'state'
 
-    def get_url(self, *args, **kwargs):
+    def get_redirect_url(self, *args, **kwargs):
```
Comment (zopieux) [inline prologin/users/auth_token_views.py:70]: Can you confirm this only called once per `get()`?

**Review by zopieux — CHANGES_REQUESTED** [review]

File: prologin/users/auth_token_views.py (hunk 1 above)
Comment (remi-dupre) [reply prologin/users/auth_token_views.py]: **mind blowed\*\*

File: prologin/users/auth_token_views.py (hunk 2 above)
Comment (remi-dupre) [reply prologin/users/auth_token_views.py:70]: It appears so: https://ccbv.co.uk/projects/Django/2.1/django.views.generic.base/RedirectView/#get

**Review by zopieux — APPROVED** [review]
Please squash with a nice commit summary.

**Pull request merged**